packages/binding-coap/src/coap-server.ts

```diff
@@ -562,6 +580,48 @@ export default class CoapServer implements ProtocolServer {
         res.end("Not Found");
     }
 
+    private parseUrlParameters(
+        url: string,
+        globalUriVariables: { [key: string]: TD.DataSchema },
+        uriVariables: { [key: string]: TD.DataSchema }
+    ): Record<string, unknown> {
+        const params: Record<string, unknown> = {};
+        if (url == null || !uriVariables) {
+            return params;
+        }
+
+        const queryparams = url.split("?")[1];
+        if (queryparams == null) {
+            return params;
+        }
+        const queries = queryparams.split("&");
+
+        queries.forEach((indexQuery: string) => {
+            const indexPair = indexQuery.split("=");
+
+            const queryKey: string = decodeURIComponent(indexPair[0]);
+            const queryValue: string = decodeURIComponent(indexPair.length > 1 ? indexPair[1] : "");
+
+            if (uriVariables[queryKey]) {
+                if (uriVariables[queryKey].type === "integer" || uriVariables[queryKey].type === "number") {
+                    // *cast* it to number
+                    params[queryKey] = +queryValue;
+                } else {
+                    params[queryKey] = queryValue;
+                }
+            } else if (globalUriVariables[queryKey]) {
+                if (globalUriVariables[queryKey].type === "integer" || globalUriVariables[queryKey].type === "number") {
+                    // *cast* it to number
+                    params[queryKey] = +queryValue;
+                } else {
+                    params[queryKey] = queryValue;
+                }
+            }
+        });
+
+        return params;
+    }
+
     private isEmpty(obj: Record<string, unknown>) {
```

Comment:
isEmpty doesn't depend on this and can be put outside the class as a simple function to avoid prototype pollution.